java/ql/lib/semmle/code/java/security/UrlForwardQuery.qll

```diff
@@ -0,0 +1,35 @@
+/** Provides a taint-tracking configuration for reasoning about URL forwarding. */
+
+import java
+import semmle.code.java.security.UrlForward
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.security.PathSanitizer
+
+/**
+ * A taint-tracking configuration for reasoning about URL forwarding.
+ */
+module UrlForwardFlowConfig implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node source) {
+    source instanceof ThreatModelFlowSource and
+    // excluded due to FPs
+    not exists(MethodCall mc, Method m | mc.getMethod() = m |
+      (
+        m instanceof HttpServletRequestGetRequestUriMethod or
+        m instanceof HttpServletRequestGetRequestUrlMethod or
+        m instanceof HttpServletRequestGetPathMethod
+      ) and
+      mc = source.asExpr()
+    )
```

Comment:
Small NIT if you want to avoid having to use parentheses:

```suggestion
    not exists(MethodCall mc, Method m |
      m instanceof HttpServletRequestGetRequestUriMethod or
      m instanceof HttpServletRequestGetRequestUrlMethod or
      m instanceof HttpServletRequestGetPathMethod
    |
      mc.getMethod() = m and
      mc = source.asExpr()
    )
```

Reply:
Updated in https://github.com/github/codeql/pull/14854/commits/121b24ea7c0e21f2032afc64b54f4de85bda9275.